app/controllers/YourSelfAssessmentChargesController.scala

```diff
@@ -33,26 +33,27 @@ import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
 import services.{ClaimToAdjustService, DateServiceInterface, WhatYouOweService}
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
-import views.html.WhatYouOwe
+import views.html.YourSelfAssessmentCharges
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class YourSelfAssessmentChargesController @Inject()(val authActions: AuthActions,
-                                     val whatYouOweService: WhatYouOweService,
-                                     val claimToAdjustService: ClaimToAdjustService,
-                                     val itvcErrorHandler: ItvcErrorHandler,
-                                     val itvcErrorHandlerAgent: AgentItvcErrorHandler,
-                                     val auditingService: AuditingService,
-                                     val dateService: DateServiceInterface,
-                                     whatYouOwe: WhatYouOwe
-                                    )(implicit val appConfig: FrontendAppConfig,
-                                      val mcc: MessagesControllerComponents,
-                                      val ec: ExecutionContext) extends FrontendController(mcc) with I18nSupport with FeatureSwitching{
-
+                                                    val whatYouOweService: WhatYouOweService,
+                                                    val claimToAdjustService: ClaimToAdjustService,
+                                                    val itvcErrorHandler: ItvcErrorHandler,
+                                                    val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                                    val auditingService: AuditingService,
+                                                    val dateService: DateServiceInterface,
+                                                    view: YourSelfAssessmentCharges
+                                                   )(implicit val appConfig: FrontendAppConfig,
+                                                     val mcc: MessagesControllerComponents,
+                                                     val ec: ExecutionContext) extends FrontendController(mcc)
+  with I18nSupport with FeatureSwitching {
 
   def handleRequest(backUrl: String,
                     itvcErrorHandler: ShowInternalServerError,
+                    isAgent: Boolean,
```

Comment:
again, no need to pass isAgent as its already defined in =>
`user: MtdItUser[_]`